### PR TITLE
Primary navigation

### PR DESCRIPTION
**Pull request by joshdarby**
## Changes

This pull request makes the following changes:

- Created new menu locations
   - Languages menu
   - Mobile sticky menu
   - Special projects secondary menu (placeholder, kind of, for #41. The expanded mobile nav includes this menu)
- Styles desktop menu
- Styles mobile sticky menu
- Styles expanded mobile menu

Desktop:
Screen Shot 2019-10-15 at 5 07 39 PM

Mobile sticky:
Screen Shot 2019-10-15 at 5 08 02 PM

Mobile expanded:
Screen Shot 2019-10-15 at 5 08 34 PM

## Why

For #40 

## Testing/Questions

Features that this PR affects:

- Navigation on desktop + mobile

Questions that need to be answered before merging:

- [x] Is this PR targeting the correct branch in this repository?
- [ ] Does it look ok at all sizes?
- [ ] Could the new menu locations be named better?

Steps to test this PR:

1. Create new menu based off the desktop menu found [here](https://drive.google.com/file/d/1VaA9xp4buM1BYG90NKVEGGTQ8jhIpJvW/view) and add it to the `Main Navigation` and `Global Navigation` menu locations
2. Create new menu based off of the english/spanish menu from the desktop menu found [here](https://drive.google.com/file/d/1VaA9xp4buM1BYG90NKVEGGTQ8jhIpJvW/view) and add it to the `Languages Menu` location
3. Create new menu based off of the mobile sticky menu found [here](https://drive.google.com/file/d/1NW8ysIysi_saWoKxNuWY5-HGOct9Gn9s/view) and add it to the `Mobile Sticky` menu location
4. Create new menu secondary projects menu based off of the expanded mobile menu [here](https://drive.google.com/file/d/163cz1Z824UnKhylmFjTY-a5JVm-8sTzL/view) (the red menu) and add it to the `Special Projects Secondary Navigation Menu` location
5. View the desktop menu
6. View the mobile sticky menu
7. View the expanded mobile menu

### CODE REVIEW

**Comment by benlk** [comment]
General question: Is this also supposed to affect the custom navigation going on at "Mapping the Future"/"Rezone"/"Zonein" series at https://citylimits.test/zonein/ ? The sticky nav on that page appears to be broken.

File: wp-content/themes/citylimits/functions.php [hunk 1]
```diff
@@ -362,3 +362,13 @@ function citylimits_newsletter_form_popover() {
...
 	return ob_get_clean();
 });
+
+/**
+ * Register custom menu locations for the main nav and sticky mobile navs
+ */
+function register_citylimits_menu_locations() {
+	register_nav_menu( 'languages-menu', __( 'Languages Menu' ) );
+	register_nav_menu( 'mobile-sticky-menu', __( 'Mobile Sticky' ) );
+	register_nav_menu( 'special-projects-secondary-menu', __( 'Special Projects Secondary Navigation Menu' ) ) ;
```
Comment (benlk) [inline wp-content/themes/citylimits/functions.php]: Do we want to deregister Largo's default menus here? https://github.com/INN/largo/blob/v0.6.4/functions.php#L218-L225

**Comment by joshdarby** [comment]
Good catch and good question. I didn't know those existed with their own custom navs. Looking into it.

**Comment by joshdarby** [comment]
@benlk https://github.com/INN/umbrella-citylimits/pull/65/commits/e7a126d5de8630dbbf7b64d26fb22c35ecab4557 updates `header-rezone.php` to use the main sticky nav instead of the sticky rezone menu and also adds the main nav menu above the series hero image.

@MirandaEcho Does this look ok or do we need to do more with it?

Screen Shot 2019-10-16 at 1 41 05 PM

File: wp-content/themes/citylimits/partials/nav-sticky.php [hunk 2]
```diff
@@ -91,80 +51,85 @@ class="input-medium appendedInputButton search-query" value="" name="s" />
...
-								'theme_location' => 'main-nav',
-								'depth'		 => 0,
-								'container'	 => false,
-								'items_wrap' => '%3$s',
-								'menu_class' => 'nav',
-								'walker'	 => new Bootstrap_Walker_Nav_Menu()
-							);
-							largo_nav_menu($args);
+						<div class="close-menu toggle-nav-bar">
+							<span class="dashicons dashicons-no-alt"></span>
```
Comment (benlk) [inline wp-content/themes/citylimits/partials/nav-sticky.php]: We should probably put an `aria-label="close menu"` here. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute

File: wp-content/themes/citylimits/functions.php (hunk 1 above)
Comment (joshdarby) [reply wp-content/themes/citylimits/functions.php]: @benlk Good idea. Updated in https://github.com/INN/umbrella-citylimits/pull/65/commits/c34aeabf7ae0b9936a67f8805846083e7515cd47

File: wp-content/themes/citylimits/partials/nav-sticky.php (hunk 2 above)
Comment (joshdarby) [reply wp-content/themes/citylimits/partials/nav-sticky.php]: Yep. Added in https://github.com/INN/umbrella-citylimits/pull/65/commits/b0f2bbe93ff23440ab0059fa308e32cc06df4942

**Comment by benlk** [comment]
How is the language-switcher supposed to work? We create menu items, sure, but how do those menu items affect the interface language or other parts of the site? Is there supposed to be a plugin that ties into those menu items?

File: wp-content/themes/citylimits/less/_navbar.less [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+                    text-transform: uppercase;
+                }
+                &:hover {
+                    color: @red;
+                    background: none;
+                }
+            }
+        }
+    }
+    &.open {
```
Comment (benlk) [inline wp-content/themes/citylimits/less/_navbar.less]: If the mobile nav is open, and the display resizes, then the menu remains open, and the sticky nav gets weird:
Screen Shot 2019-10-16 at 14 12 43

**Comment by joshdarby** [comment]
> How is the language-switcher supposed to work? We create menu items, sure, but how do those menu items affect the interface language or other parts of the site? Is there supposed to be a plugin that ties into those menu items?

@benlk [Here's](https://github.com/INN/umbrella-citylimits/issues/40#issuecomment-540623803) what @MirandaEcho had to say about the language switcher: 
> on the current site, the Espanol section links to the series that's in Spanish. We'll do the same with the toggle for now.

File: wp-content/themes/citylimits/less/_navbar.less (hunk 3 above)
Comment (joshdarby) [reply wp-content/themes/citylimits/less/_navbar.less]: Ah, gross. I guess we can just hide it after a certain width. Or did you have something else in mind, @benlk?

Comment (benlk) [reply wp-content/themes/citylimits/less/_navbar.less]: That's probably easier than an on-resize handler that optionally closes the menu if it's open.

**Comment by joshdarby** [comment]
@benlk https://github.com/INN/umbrella-citylimits/pull/65/commits/4753528e70aad0e92c05b9ac7027fabb425f5427 should solve the menu icon issue and https://github.com/INN/umbrella-citylimits/pull/65/commits/2fdf290d3cc5dab49d13e24d5ca02387de418055 should solve the scrolling after resize issue

**Pull request merged**